internal/cmdrender/runtime/container.go

```diff
@@ -66,12 +73,28 @@ func (f *ContainerFn) Run(reader io.Reader, writer io.Writer) error {
 	cmd.Stdout = writer
 	cmd.Stderr = &errSink
 
-	pr.PkgPrintf(f.Path, "running function %q: ", f.Image)
+	printOpt := printer.NewOpt().WithIndentation(printer.FnIndentation)
+	pr.Printf(printOpt, "[RUNNING] %q\n", f.Image)
```

Comment:
should hardcode presentation-related things at this layer. Presentation should also be shared with `eval`.

Extract this out.

Reply:
@Shell32-Natsu  Note that `ContainerFn` needs to implement interface with `Run(r,w)`, so move the function execution logic to an internal function and `Run` should wrap that internal function. Rough skeleton:

```
func (f *ContainerFn) Run(r io.Reader, w io.Writer) error {

   // printer stuff (mostly saying function is RUNNING) (That prepareImage thing is going to be an issue here)
   fnExecError := f.internalRun(r, w)
  // printer stuff related to post function run. Note fnExecError is printable
}

```

Complete decoupling where we spin off the function running in a separate go rountine and get the progress on a channel on the main goroutine that prints the progress can be done, but will be lot more work.

Reply:
Added a container runner wrapper which will print these progress information. In future `fn eval` can reuse the same wrapper.